src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionOptions.cs

```diff
@@ -105,5 +107,87 @@ public bool ContainsKey(string keyword)
         {
             return _parsetable.ContainsKey(keyword);
         }
+
+        protected internal virtual string Expand()
+        {
+            return _usersConnectionString;
+        }
+
+        // SxS notes:
+        // * this method queries "DataDirectory" value from the current AppDomain.
+        //   This string is used for to replace "!DataDirectory!" values in the connection string, it is not considered as an "exposed resource".
+        // * This method uses GetFullPath to validate that root path is valid, the result is not exposed out.
+        internal static string ExpandDataDirectory(string keyword, string value)
+        {
+            string fullPath = null;
+            if ((null != value) && value.StartsWith(DataDirectory, StringComparison.OrdinalIgnoreCase))
+            {
+                // find the replacement path
+                object rootFolderObject = AppDomain.CurrentDomain.GetData("DataDirectory");
+                var rootFolderPath = (rootFolderObject as string);
+                if ((null != rootFolderObject) && (null == rootFolderPath))
+                {
+                    throw ADP.InvalidDataDirectory();
+                }
+                else if (string.IsNullOrEmpty(rootFolderPath))
+                {
+                    rootFolderPath = AppDomain.CurrentDomain.BaseDirectory;
+                }
+                if (null == rootFolderPath)
+                {
+                    rootFolderPath = "";
+                }
```

Comment:
This could be modernized to `rootFolderPath = AppDomain.CurrentDomain.BaseDirectory ?? string.Empty;`

Reply:
Done